packages/std/src/math/int128.rs

```diff
@@ -0,0 +1,1106 @@
+    #[inline]
+    pub fn wrapping_pow(self, other: u32) -> Self {
+        Self(self.0.wrapping_pow(other))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_add(self, other: Self) -> Self {
+        Self(self.0.saturating_add(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_sub(self, other: Self) -> Self {
+        Self(self.0.saturating_sub(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_mul(self, other: Self) -> Self {
+        Self(self.0.saturating_mul(other.0))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn saturating_pow(self, exp: u32) -> Self {
+        Self(self.0.saturating_pow(exp))
+    }
+
+    #[must_use = "this returns the result of the operation, without modifying the original"]
+    pub fn abs_diff(self, other: Self) -> Uint128 {
+        Uint128(self.0.abs_diff(other.0))
+    }
+}
+
+impl From<Uint64> for Int128 {
+    fn from(val: Uint64) -> Self {
+        val.u64().into()
+    }
+}
+
+impl From<u64> for Int128 {
+    fn from(val: u64) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<u32> for Int128 {
+    fn from(val: u32) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<u16> for Int128 {
+    fn from(val: u16) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<u8> for Int128 {
+    fn from(val: u8) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<i128> for Int128 {
+    fn from(val: i128) -> Self {
+        Int128(val)
+    }
+}
+
+impl From<i64> for Int128 {
+    fn from(val: i64) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<i32> for Int128 {
+    fn from(val: i32) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<i16> for Int128 {
+    fn from(val: i16) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl From<i8> for Int128 {
+    fn from(val: i8) -> Self {
+        Int128(val.into())
+    }
+}
+
+impl TryFrom<&str> for Int128 {
+    type Error = StdError;
+
+    fn try_from(val: &str) -> Result<Self, Self::Error> {
+        Self::from_str(val)
+    }
+}
+
+impl FromStr for Int128 {
+    type Err = StdError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.parse::<i128>() {
+            Ok(u) => Ok(Self(u)),
+            Err(e) => Err(StdError::generic_err(format!("Parsing Int128: {}", e))),
+        }
+    }
+}
+
+impl From<Int128> for String {
+    fn from(original: Int128) -> Self {
+        original.to_string()
+    }
+}
+
+impl fmt::Display for Int128 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // The inner type doesn't work as expected with padding, so we
+        // work around that. Remove this code when the upstream padding is fixed.
```

Comment:
Is this true for i128? Or would it be better to use bum for the implementation here to get consistency?

Reply:
Oh, good catch. This is only true for bnum.